src/compiler/checker.ts

```diff
@@ -14065,6 +14044,126 @@ namespace ts {
+                    if (intrinsicElementsType !== unknownType) {
+                        const stringLiteralTypeName = (<StringLiteralType>valueType).value;
+                        const intrinsicProp = getPropertyOfType(intrinsicElementsType, escapeLeadingUnderscores(stringLiteralTypeName));
+                        if (intrinsicProp) {
+                            return getTypeOfSymbol(intrinsicProp);
+                        }
+                        const indexSignatureType = getIndexTypeOfType(intrinsicElementsType, IndexKind.String);
+                        if (indexSignatureType) {
+                            return indexSignatureType;
+                        }
+                    }
+                    return anyType;
+                }
+
+                // Resolve the signatures, preferring constructor
+                let signatures = getSignaturesOfType(valueType, SignatureKind.Construct);
+                let ctor = true;
+                if (signatures.length === 0) {
+                    // No construct signatures, try call signatures
+                    signatures = getSignaturesOfType(valueType, SignatureKind.Call);
+                    ctor = false;
+                    if (signatures.length === 0) {
+                        // We found no signatures at all, which is an error
+                        return unknownType;
+                    }
+                }
+
+                return getUnionType(map(signatures, ctor ? getJsxPropsTypeFromConstructSignature : getJsxPropsTypeFromCallSignature), UnionReduction.None);
+            }
+        }
+
+        function getJsxPropsTypeFromCallSignature(sig: Signature) {
+            let propsType = getTypeOfFirstParameterOfSignature(sig);
+            const intrinsicAttribs = getJsxType(JsxNames.IntrinsicAttributes);
+            if (intrinsicAttribs !== unknownType) {
+                propsType = intersectTypes(intrinsicAttribs, propsType);
+            }
+            return propsType;
+        }
+
+        function getJsxPropsTypeFromClassType(hostClassType: Type) {
+            if (isTypeAny(hostClassType)) {
+                return hostClassType;
+            }
+
+            const propsName = getJsxElementPropertiesName();
+            if (propsName === undefined) {
+                // There is no type ElementAttributesProperty, return 'any'
+                return anyType;
+            }
+            else if (propsName === "") {
+                // If there is no e.g. 'props' member in ElementAttributesProperty, use the element class type instead
+                return hostClassType;
+            }
+            else {
+                const attributesType = getTypeOfPropertyOfType(hostClassType, propsName);
+
+                if (!attributesType) {
+                    // There is no property named 'props' on this instance type
+                    return emptyObjectType;
+                }
+                else if (isTypeAny(attributesType)) {

```

Comment:
Why drop the unknownType check? Or does isTypeAny work for unknownType too?

Reply:
Yes. @RyanCavangauh pointed that out.